state-chain/chains/src/evm/benchmarking.rs

```diff
@@ -1,19 +1,66 @@
 #![cfg(feature = "runtime-benchmarks")]
 
-use super::TransactionFee;
+use super::{
+	api::{EvmEnvironmentProvider, EvmReplayProtection},
+	TransactionFee,
+};
 use crate::{
 	benchmarking_value::{BenchmarkValue, BenchmarkValueExtended},
 	eth::api::EthereumApi,
-	evm::{to_evm_address, Address, AggKey, SchnorrVerificationComponents, Transaction, H256},
-	ApiCall,
+	evm::{api, to_evm_address, Address, AggKey, SchnorrVerificationComponents, Transaction, H256},
+	ApiCall, Chain, ReplayProtectionProvider,
+};
+use cf_primitives::{
+	chains::{assets, Ethereum},
+	EthAmount,
 };
-use cf_primitives::EthAmount;
 use ethabi::Uint;
 use libsecp256k1::{PublicKey, SecretKey};
 
 const SIG_NONCE: [u8; 32] = [1u8; 32];
 const PRIVATE_KEY: [u8; 32] = [2u8; 32];
 
+pub struct EvmBenchmarkEnv;
+
+impl<C: Chain<ReplayProtection = EvmReplayProtection, ReplayProtectionParams = Address>>
+	ReplayProtectionProvider<C> for EvmBenchmarkEnv
+where
+	EvmBenchmarkEnv: EvmEnvironmentProvider<C>,
+{
+	fn replay_protection(
+		contract_address: <C as Chain>::ReplayProtectionParams,
+	) -> EvmReplayProtection {
+		EvmReplayProtection {
+			nonce: <Self as EvmEnvironmentProvider<C>>::next_nonce(),
+			chain_id: <Self as EvmEnvironmentProvider<C>>::chain_id(),
+			key_manager_address: <Self as EvmEnvironmentProvider<C>>::key_manager_address(),
+			contract_address,
+		}
+	}
+}
+
+impl EvmEnvironmentProvider<Ethereum> for EvmBenchmarkEnv {
+	fn token_address(_asset: assets::eth::Asset) -> Option<Address> {
+		unimplemented!()
+	}
+
+	fn vault_address() -> Address {
+		unimplemented!()
+	}
+
+	fn key_manager_address() -> Address {
+		unimplemented!()
+	}
+
+	fn chain_id() -> api::EvmChainId {
+		unimplemented!()
+	}
+
+	fn next_nonce() -> u64 {
+		unimplemented!()
```

Comment:
Will this not cause the benchmarks to panic? 

Should we not return `BenchmarkValue::benchmark_value()` for each of these instead of `unimplemented!()`?

Reply:
No - I tried it. But yeah, I agree, returning a proper value makes sense.